pkg/kubelet/cm/memorymanager/memory_manager.go

```diff
@@ -182,6 +182,14 @@ func (m *manager) AddContainer(pod *v1.Pod, container *v1.Container, containerID
 	defer m.Unlock()
 
 	m.containerMap.Add(string(pod.UID), container.Name, containerID)
+
+	if isInitContainer(pod, container) {
+		return
+	}
+
+	// if the app container started and pod has init containers we should remove it
+	// to release memory requested by init containers
+	m.removeInitContainers(pod)
```

Comment:
Why does this need to be special cased for `app` containers. I would expect an init container to also want to remove the memory of previous init containers in the sequence before having memory allocated to it.

Reply:
The allocation for init containers and the first app will always use a saved state that will not include init container allocations. Also, be aware that `AddContainer` will run after all pod containers allocations, so it can not affect in any way pod containers allocations.
But anyway it can be a good thing to remove init containers allocations from the state file quicker because it will release memory for other containers.